### PR TITLE
bank_branch: add implementation

### PR DESCRIPTION
**Pull request by N1cOs**

### CODE REVIEW

File: src/bank_branch.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  if (len > 0) {
+    BalanceState prev = history->s_history[len - 1];
+    timestamp_t elapsed = state.s_time - prev.s_time - 1;
+    for (int i = 0; i < elapsed; i++) {
+      history->s_history[len + i] = prev;
+    }
+    len += elapsed;
+  }
+  history->s_history[len] = state;
+  history->s_history_len++;
```
Comment (dmitriyace) [inline src/bank_branch.c]: Please, don't forget to add elapsed time here.

File: src/bank_branch.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  new_state.s_time = get_physical_time();
+  new_state.s_balance += delta;
+  _append_balance_state(history, new_state);
+  return new_state;
+}
+
+static int _sync(BankBranch *branch, Message *msg) {
+  if (send_multicast(branch->client, msg) != 0) {
+    fprintf(stderr, "error: branch %1d: %s: %s\n", branch->id, "send_multicast",
+            "start");
```
Comment (dmitriyace) [inline src/bank_branch.c]: Please, be more careful

Comment (dmitriyace) [reply src/bank_branch.c]:
```suggestion
            "sync");
```

File: src/bank_branch.c [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                  "receive_any", "bad message recipient");
+          return EXIT_FAILURE;
+        }
+        break;
+      }
+      case STOP:
+        break;
+      default:
+        fprintf(stderr, "error: branch %1d: %s: %s\n", branch->id,
+                "receive_any", "unknown message type");
```
Comment (dmitriyace) [inline src/bank_branch.c]: What do you mean under 'unknown'? `unexpected` sounds much more suitable here.

File: src/bank_branch.c [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  MessageHeader balance_header;
+  balance_header.s_magic = MESSAGE_MAGIC;
+  balance_header.s_type = BALANCE_HISTORY;
+  balance_header.s_local_time = get_physical_time();
+  balance_header.s_payload_len = sizeof(BalanceHistory);
+
+  Message balance_msg = {balance_header};
+  memcpy(balance_msg.s_payload, branch->history, balance_header.s_payload_len);
+
+  if (send(branch->client, PARENT_ID, &msg) != 0) {
```
Comment (dmitriyace) [inline src/bank_branch.c]: Most probably you wanted to write `balance_msg` here.

**Review by dmitriyace — CHANGES_REQUESTED** [review]

File: src/bank_branch.c (hunk 1 above)
Comment (N1cOs) [reply src/bank_branch.c]: fixed

**Review by dmitriyace — APPROVED** [review]

**Pull request merged**